src/main/java/com/minecolonies/coremod/permissions/ColonyPermissionEventHandler.java

```diff
@@ -206,6 +205,57 @@ public void on(final PlayerInteractEvent event)
     }
 
     /**
+     * PlayerInteractEvent.EntityInteract handler.
+     * <p>
+     * Check, if a player right clicked an entity.
+     * Deny if:
+     * - If the entity is in colony
+     * - player has not permission
+     *
+     * @param event PlayerInteractEvent
+     */
+    @SubscribeEvent
+    public void on(final PlayerInteractEvent.EntityInteract event)
+    {
+        final EntityPlayer playerIn = event.getEntityPlayer();
+        if (Configurations.enableColonyProtection && colony.isCoordInColony(playerIn.getEntityWorld(), playerIn.getPosition()))
+        {
+            final Permissions.Rank rank = colony.getPermissions().getRank(playerIn);
+
+            if (rank.ordinal() >= Permissions.Rank.FRIEND.ordinal())
```

Comment:
ACCESS_HUTS isn't a rank, it's a permission. I suppose if we load enums by ordinal, I can see how an issue would arise when we make new ones. We can get around this, by not changing the order of the enums. You should be free to create new actions/permissions if you add them to the end of the enum.